db/migrate/20150629075221_add_scm_type_to_repositories.rb

```diff
@@ -0,0 +1,54 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+##
+# Adds the scm_type column to the repositories table and updates all previous
+# repository to correspond to these types.
+#
+# As until now, OP only supported local Git repositories and local or remote Subversion
```

Comment:
That is to differentiate between existing, but local-only repositories (Git does not allow remote repositories currently, and will not after this PR), and Subversion's URL-based existing repositories (which doesn't care whether the repository is accessed on the filesystem, or on a remote server).

I could have named the scm_type for Git existing as well, but the reason I chose not to was to make some later wishlist upgrade [that allows remote Git repositories](https://community.openproject.org/work_packages/20422) to change that type back to existing for consistency.

In the end, all that this type controls is the partial that is rendered from /repositories/settings/<vendor>/<type>. It only does more when it is `:managed`.